trafficpolicy/policy.go

```diff
@@ -0,0 +1,91 @@
+package trafficpolicy
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Policy struct {
+	// the ordered set of rules that apply to inbound traffic
+	Inbound []Rule `json:"inbound,omitempty" yaml:"inbound,omitempty"`
+	// the ordered set of rules that apply to outbound traffic
+	Outbound []Rule `json:"outbound,omitempty" yaml:"outbound,omitempty"`
+}
+
+type Rule struct {
+	// the name of the traffic policy rule
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+	// the set of CEL expressions used to determine if this rule is applicable
+	Expressions []string `json:"expressions,omitempty" yaml:"expressions,omitempty"`
+	// the ordered set of actions that should take effect against the traffic
+	Actions []Action `json:"actions" yaml:"actions"`
+}
+
+type Action struct {
+	// the type of action that should be used
+	Type string `json:"type" yaml:"type"`
+	// the configuration for the specified action type written as a json string
+	Config ActionConfigString `json:"config,omitempty" yaml:"config,omitempty"`
+}
+
+type ActionConfigString string
+
+func (s ActionConfigString) MarshalJSON() ([]byte, error) {
+	raw := json.RawMessage([]byte(s))
+
+	return json.Marshal(raw)
+}
+
+func (cfg *ActionConfigString) UnmarshalJSON(data []byte) error {
+	s := ActionConfigString(data)
+	*cfg = s
+	return nil
+}
+
+func (p Policy) MarshalJSON() (string, error) {
+	return marshalJSON(p)
+}
+
+func (p Rule) MarshalJSON() (string, error) {
+	return marshalJSON(p)
+}
+
+func (p Action) MarshalJSON() (string, error) {
+	return marshalJSON(p)
+}
```

Comment:
I actually don't think naming these `MarshalJSON()` here is the right move after all since that's the function name used to override a type's marshaling behavior (see what I mean above for ActionConfigString). Leaning towards keeping these as `ToJSON()` or even just `JSON()`

Reply:
Does that change consistency with the other parts of the API? Are there some places where it's called `MarshalJSON` and others where it is `ToJSON`? That would be my only concern. If not, then I'm fine calling it `ToJSON` or `JSON`.

Reply:
It would be consistent to the user of this package to always use `ToJSON` or `JSON` — the custom type `ActionConfigString` would never have its `MarshalJSON()` or `MarshalYAML()` function called except by the json/yaml encoders

Reply:
We landed on `JSON()` and `YAML()`